Lib/http/server.py

```diff
@@ -280,9 +280,9 @@ def parse_request(self):
         self.request_version = version = self.default_request_version
         self.close_connection = True
         requestline = str(self.raw_requestline, 'iso-8859-1')
-        requestline = requestline.rstrip('\r\n')
+        requestline = requestline.rstrip(' \r\n')
```

Comment:
Why do you strip trailing spaces?

Reply:
Leaving it as `rstrip('\r\n')` would cause test_version_none_get to fail:

```
FAIL: test_version_none_get (Lib.test.test_httpservers.BaseHTTPServerTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../cpython/Lib/test/test_httpservers.py", line 167, in test_version_none_get
    self.assertEqual(res.status, HTTPStatus.NOT_IMPLEMENTED)
AssertionError: 400 != <HTTPStatus.NOT_IMPLEMENTED: 501>
```

Maybe the 400 would be better, though?

Reply:
I see. But you should not modify requestline.